### PR TITLE
WT-7519 Fix flags field overflow in WT_DATA_HANDLE

### PR DESCRIPTION
**Pull request by keitharnoldsmith**
Repartition division of flags between WT_DATA_HANDLE and WT_BTREE.

WT_DATA_HANDLE now has 12 available flags and WT_BTREE has 20.

Also added a max flag (WT_DHANDLE_ZZZ_ENDFLAG) that we can use to assert
that the dhandle flags haven't grown into the space reserved for btree
flags.

### CODE REVIEW

File: dist/s_string.ok [hunk 1]
```diff
@@ -1493,6 +1493,7 @@ wtstats
 xF
 xdeadbeef
 xff
+xfff
```
Comment (sueloverso) [inline dist/s_string.ok:1496]: Can you grep for xff? Do we still need that?

File: src/include/dhandle.h [hunk 2]
```diff
@@ -104,7 +104,14 @@ struct __wt_data_handle {
...
     WT_DSRC_STATS *stat_array;
 
-/* Flags values over 0xff are reserved for WT_BTREE_* */
+/*
+ * Flags values over 0xfff are reserved for WT_BTREE_*. This lets us combine the dhandle and btree
+ * flags when we need, for example, to pass both sets in a function call.
+ *
+ * To help avoid accidental overrun of the flag values, we add a special flag value that should
+ * always be the last and highest. At runtime, we use this value to assert that the dhandle flags
+ * haven't run into the space reserved for btree flags.
```
Comment (sueloverso) [inline src/include/dhandle.h:113]: If you use the preprocessor, change the comment to not say 'at runtime'.

**Review by sueloverso — APPROVED** [review]
I prefer the preprocessor. I'm happy with whichever way you decide to go.

**Comment by keitharnoldsmith** [comment]
I also prefer the preprocessor approach.  I just wasn't sure that was a thing we do in WT code.  I could have used `grep` to check, but wasn't thinking...  I'll change to do it that way.

File: src/include/dhandle.h (hunk 2 above)
Comment (keitharnoldsmith) [reply src/include/dhandle.h]: Done

**Pull request merged**